runtime/js-compute-runtime/builtins/cache-simple.cpp

```diff
@@ -0,0 +1,440 @@
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+  if (!args.requireAtLeast(cx, "SimpleCache.get", 1)) {
+    return false;
+  }
+
+  fastly_world_string_t key;
+  // Convert key parameter into a string and check the value adheres to our validation rules.
+  JS::UniqueChars key_chars = encode(cx, args[0], &key.len);
+  if (!key_chars) {
+    return false;
+  }
+  key.ptr = key_chars.get();
+
+  if (key.len == 0) {
+    JS_ReportErrorASCII(cx, "SimpleCache.get: key can not be an empty string");
+    return false;
+  }
+  if (key.len > 8135) {
+    JS_ReportErrorASCII(cx,
+                        "SimpleCache.get: key is too long, the maximum allowed length is 8135.");
+    return false;
+  }
+
+  fastly_error_t err;
+  fastly_cache_lookup_options_t options;
+  std::memset(&options, 0, sizeof(options));
+
+  fastly_cache_handle_t handle = INVALID_HANDLE;
+  if (!fastly_cache_lookup(&key, &options, &handle, &err)) {
+    HANDLE_ERROR(cx, err);
+    return false;
+  }
+
+  fastly_body_handle_t body = INVALID_HANDLE;
+  fastly_cache_get_body_options_t opts;
+  if (!fastly_cache_get_body(handle, &opts, &body, &err)) {
+    HANDLE_ERROR(cx, err);
+    return false;
+  }
+
+  if (body == INVALID_HANDLE) {
+    args.rval().setNull();
+  } else {
+    JS::RootedObject entry(cx, SimpleCacheEntry::create(cx, body));
+    if (!entry) {
+      return false;
+    }
+    args.rval().setObject(*entry);
+  }
+
+  return true;
+}
+
+// static delete(key: string): undefined;
+bool SimpleCache::delete_(JSContext *cx, unsigned argc, JS::Value *vp) {
+  REQUEST_HANDLER_ONLY("The SimpleCache builtin");
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+  if (!args.requireAtLeast(cx, "SimpleCache.delete", 1)) {
+    return false;
+  }
+
+  fastly_world_string_t key;
+  // Convert key parameter into a string and check the value adheres to our validation rules.
+  {
+    JS::UniqueChars key_chars = encode(cx, args.get(0), &key.len);
+    if (!key_chars) {
+      return false;
+    }
+    key.ptr = key_chars.get();
+
+    if (key.len == 0) {
+      JS_ReportErrorASCII(cx, "SimpleCache.delete: key can not be an empty string");
+      return false;
+    }
+    if (key.len > 1024) {
+      JS_ReportErrorASCII(
+          cx, "SimpleCache.delete: key is too long, the maximum allowed length is 1024.");
+      return false;
+    }
+  }
+
+  // We create a surrogate-key from the cache-key, as this allows the cached contents to be purgable
+  // from within the JavaScript application
+  // This is because the cache API currently only supports purging via surrogate-key
+  auto surrogate_key_result = createSurrogateKeyFromCacheKey(cx, key);
+  if (surrogate_key_result.isErr()) {
+    return false;
+  }
+  auto surrogate_key = surrogate_key_result.unwrap();
+
+  fastly_world_string_t skey;
+  skey.ptr = const_cast<char *>(surrogate_key.c_str());
+  skey.len = surrogate_key.length();
+
+  fastly_error_t err;
+  fastly_world_option_string_t ret;
```

Comment:
We need to free the string pointed to by `ret.val` if `ret.is_some` is true, if the call to `fastly_purge_surrogate_key` is successful.